server/lib/flow-event.js

```diff
@@ -32,26 +124,17 @@ module.exports = function (event, data, request) {
     hostname: HOSTNAME,
     op: 'flowEvent',
     pid: process.pid,
-    time: event.time,
+    time: new Date(event.time).toISOString(),
     userAgent: request.headers['user-agent'],
     v: VERSION
   }, _.mapValues(pickedData, sanitiseData));
 
   optionallySetFallbackData(eventData, 'service', data.client_id);
   optionallySetFallbackData(eventData, 'entrypoint', data.entryPoint);
 
-  if (typeof eventData.time === 'number') {
-    eventData.time = new Date(eventData.time).toISOString();
-  }
-
-  return new Promise(function (resolve) {
-    setImmediate(function () {
```

Comment:
I realised this `setImmediate` was redundant because the route handler already wraps all the logging inside a `process.nextTick`.